Replace `templatemap_generator.php` with a `Symfony\Console` command

gsteel opened the pull request:
Closes #218

gsteel committed:
Initial rev of a template map generator class.

This class will generate the necessary PHP code by scanning a directory for template files.

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Remove the old standalone template map generator

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Initial rev of a symfony cli command to generate template maps

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Restrict deprecations to our own source code

Signed-off-by: George Steel <george@net-glue.co.uk>

Xerkus commented:
Package does not require symfony console. Symfony had breaking changes incompatible with existing commands just recently. Imo this should be in a satellite package with explicit dependencies.

gsteel commented:
> Package does not require symfony console. Symfony had breaking changes incompatible with existing commands just recently. Imo this should be in a satellite package with explicit dependencies

I don't disagree.

I also hate having symbols used in `src` that are not dependencies in `require`.

There's very little point going to all the trouble of setting up another repo for this simple task though. We'd end up making a repo full of utility console commands that would have to drag in dependencies like `laminas-view` for a single command even if the project wasn't using it.

As a user, who installs laminas-view and also laminas-cli, this provides a 'batteries included' utility IMO that is more important than the pain of `symfony-console` as a dev dependency.

gsteel commented:
@Xerkus - there is also more discussion in #218

froschdesign commented:
> Imo this should be in a satellite package with explicit dependencies.

That is a spiral: we have more and more packages to manage, so I would approach this with caution. It might be possible to create a package for commands that contains everything across components but only for development.

I personally like the pragmatical solution like in https://github.com/laminas/laminas-view/issues/218#issuecomment-3252209445.

gsteel committed:
Conditionally register CLI commands based on whether `symfony/cli` is installed.

This will prevent issues for users that automate integration tests for their entire DI container configuration.

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel commented:
@Xerkus and @froschdesign I've added a conditional so that CLI config is only exposed for users that have installed symfony/cli.

Can we put this one to bed? It'd be nice to get v3 released 😁

froschdesign reviewed (approved):
I am fine with this! 👍🏻
